frontend/packages/ux-editor/src/components/AddOption.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { useState } from 'react';
+import classes from './AddOption.module.css';
+import { TextResource } from './TextResource';
+import { ButtonColor, ButtonVariant, TextField } from '@altinn/altinn-design-system';
+import { Button, FieldSet } from '@digdir/design-system-react';
+import { IGenericEditComponent } from './config/componentConfig';
+import { IFormCheckboxComponent, IFormRadioButtonComponent, IOptions } from '../types/global';
+import { Add } from '@navikt/ds-icons';
+import { generateRandomId } from 'app-shared/utils/generateRandomId';
+import { useText } from '../hooks';
+import { addOptionToComponent } from '../utils/component';
+
+export interface AddOptionProps extends IGenericEditComponent {
+  addButtonClass: string;
+  component: IFormCheckboxComponent | IFormRadioButtonComponent;
+}
+
+const initialNewOption = () => ({ label: generateRandomId(12), value: generateRandomId(4) });
```

Comment:
Hva tenker du om å gjøre denne om til et objekt istedenfor en funksjon? :) Hvis du endrer på dette, så må vi huske å endre de stedene denne funksjonen kalles.

```suggestion
const initialNewOption = { label: generateRandomId(12), value: generateRandomId(4) }
```

Reply:
Hadde brukeropplevelsen blitt bedre om vi unngikk å generere en random value? Heller ha et tomt felt som viser en placeholder?

Reply:
Jeg har valgt å gjøre det til en funksjon for å være sikker på at den genererer en ny verdi hver gang den kalles.
Når det gjelder brukeropplevelsen, er jeg usikker. På datamodelleringssidene er mye gjort på tilsvarende måte for at brukerne skal kunne legge til så mange felter de vil først, og fylle dem ut etterpå (og man må jo ha unike verdier for at det skal være gyldig). Det er derfor jeg har gjort dette her. Vi kan sikkert lage en løsning om fungerer uten at brukeren har oppgitt verdier, men da tror jeg det er mye vi må endre på.

Reply:
Takk for god tilbakemelding, tenker vi beholder det slik som du har gjort det! :)